Fix class scanning forcing classes as beans

kdubb opened the pull request:
Due to _some_ discovered classes bean added directly to `Weld::addBeanClass` they would be treated as CDI beans with default scopes and qualifiers; even when they were not supposed to be considered beans.

Now only classes added via the `AddBeanClasses`,  `AddEnabledInterceptors` and `AddEnabledDecorators` annotations are forcably treated as beans.  All other classes must contain a bean defining annotation (as described by spec 2.0)

manovotn commented on junit5/src/main/java/org/jboss/weld/junit5/auto/ClassScanning.java:
```diff
@@ -255,14 +253,16 @@ private static Extension createExtension(Class<? extends Extension> clazz) {
-                        AnnotationUtils.isAnnotated(clazz, RequestScoped.class) ||
-                        AnnotationUtils.isAnnotated(clazz, Dependent.class) ||
-                        AnnotationUtils.isAnnotated(clazz, Stereotype.class);
+        return isAnnotated(clazz, NormalScope.class) || isAnnotated(clazz, Scope.class) ||
```
`isAnnotated(clazz, Scope.class)` needs to check for just `Dependent` because no other non-normal scope annotation is considered bean defining according to specification.

manovotn commented on junit5/src/main/java/org/jboss/weld/junit5/auto/ClassScanning.java:
```diff
@@ -255,14 +253,16 @@ private static Extension createExtension(Class<? extends Extension> clazz) {
 
+    private static boolean isBeanParameterAnnotation(Annotation ann) {
+        return isAnnotated(ann.annotationType(), Qualifier.class) ||
+                isAnnotated(ann.annotationType(), Scope.class) ||
```
If I get you right here, you use this to check method parameters.
In that case, the parameters need to have qualifiers to be considered beans for injection. You will never have a scope there.

kdubb commented:
@manovotn I've added the tests and reworked annotation identification per your review.

I reordered the commits so you can see easily reset HEAD to isolate the tests and see the failures, which are then fixed by the following commit.

manovotn commented:
I've create a tracking issue at #77 
Quickscanning the PR, it looks good, I'll inspect it closer later today or tomorrow once I get over other stuff.

> I reordered the commits so you can see easily reset HEAD to isolate the tests and see the failures

Thanks, though the reason I want tests is not just for me to play with them. It is to prevent future regressions ;-)

manovotn commented on junit5/src/test/java/org/jboss/weld/junit5/auto/ScannedClassesAreNotForcedBeansTest.java:
```diff
@@ -0,0 +1,39 @@
+
+
+@EnableAutoWeld
+@AddBeanClasses(ConstructedV8.class)
```
Wasn't this supposed to be `@AddBeanClasses(InjectedV8.class)` ?
Since as it stand, `ScannedParameterClassesAreNotForcedBeansTest` does exactly the same thing.

kdubb replied:
Correct. Good catch!

manovotn commented:
Thanks for contribution!
